FLT/DedekindDomain/FiniteAdeleRing/BaseChange.lean

```diff
@@ -436,11 +436,46 @@ variable [Algebra K (FiniteAdeleRing B L)]
 noncomputable def FiniteAdeleRing.baseChange : FiniteAdeleRing A K →ₐ[K] FiniteAdeleRing B L where
   toFun ak := ⟨ProdAdicCompletions.baseChange A K L B ak.1,
     (ProdAdicCompletions.baseChange_isFiniteAdele_iff A K L B ak).1 ak.2⟩
-  map_one' := sorry
-  map_mul' := sorry
-  map_zero' := sorry
-  map_add' := sorry
-  commutes' := sorry
+  map_one' := by
+    refine ext B L ?_
+    have h : (1 : FiniteAdeleRing A K) = (1 : ProdAdicCompletions A K) := rfl
```

Comment:
This is really annoying and it's not your fault.

My instinct is that this should be a lemma `coe_one` in Mathlib, around line 340 of `RingTheory.DedekindDomain.FiniteAdeleRing`, and similarly there should be lemmas `coe_zero`, `coe_add` and `coe_mul`, proved with `rfl`. This would make the proof here tidier because you'd just be able to `rw` the proofs and finish the entire proof here in one line. Similarly you'd be able to tidy up all your other proofs in this PR. I'm happy to merge this PR with these lines in, but I would encourage you to make a PR to mathlib adding these basic lemmas because you won't be the only person who needs them.